### DISCUSSION TITLE
Cron scheduled tasks cleanup on TaskiqScheduler startup

### DISCUSSION THREAD

**Discussion by sector119** [taskiq-python/taskiq#295] [General]
Hello,

After scheduler restart all cron scheduled tasks are getting duplicated...
Should we remove all scheduled tasks on scheduler startup like in code bellow? Right now only time (not cron) scheduled tasks are removed on "post_send", but if scheduler was killed or stopped before task was delivered to the worker it wouldn't be deleted too

```
async def startup(self) -> None:
    for source in self.sources:
        tasks = await source.get_schedules()

        for task in tasks:
            await source.delete_schedule(task.schedule_id)
```

**Comment by s3rius** [comment]
What schedulers are you using? Because this behavior is unintended and seems like it's a more of a bug.

**Reply by sector119** [reply]
TaskiqScheduler with RedisScheduleSource

**Reply by s3rius** [reply]
Basically, the logic is simple. The schedule source itself doesn't populate tasks on startup. Also, since cron-tasks don't have any due date it's up to people to maintain their state. 

I'm not sure why you get duplicates, even if you don't create something on startup. If you need to declare any cron tasks in code, don't use redisschedulesource for that use label schedule source instead. You can combine multiple sources in one TaskiqScheduler.
